packages/schematics/angular/component/index.ts

```diff
@@ -49,15 +50,15 @@ function addDeclarationToNgModule(options: ComponentOptions): Rule {
       return host;
     }
 
-    options.type = !!options.type ? options.type : 'Component';
+    options.type = options.type != null ? options.type : 'Component';
 
     const modulePath = options.module;
     const source = readIntoSourceFile(host, modulePath);
 
     const componentPath = `/${options.path}/`
```

Comment:
Since this line is already calculating the file name base, I think we should consider splitting this into two pieces and using them on the templates.  “componentPath” can be the first two parts of the current.  And “componentFile” can be the remainder as the filename part.  This second part can then be used has the base for all file elements in the templates.
This would remove the need to fix the broken file names with a rename as well as optimize the name creation so that it doesn’t have to happen in multiple places as it does currently.

Reply:
This componentPath we are using to add component in appModule. We are not using it for template currently.